Realm/Realm/Linq/RealmResults.cs

```diff
@@ -70,9 +72,23 @@ internal override CollectionHandleBase CreateHandle()
 
         #region IList members
 
-        public override bool Contains(object value) => Contains((T)value);
+        public override int IndexOf(T value)
+        {
+            Argument.NotNull(value, nameof(value));
+
+            if (_argumentType != (PropertyType.Object | PropertyType.Nullable))
+            {
+                throw new NotSupportedException("IndexOf on non-object results is not supported.");
+            }
 
-        public override int IndexOf(object value) => IndexOf((T)value);
+            var obj = Operator.Convert<T, RealmObject>(value);
+            if (!obj.IsManaged)
```

Comment:
Shouldn't we rather check instead that the object belongs to the same realm, not that it belongs to a realm?